attestation-service/attestation-service/src/bin/restful-as.rs

```diff
@@ -79,27 +99,31 @@ async fn main() -> Result<()> {
 
     let server = match (cli.https_prikey, cli.https_pubkey_cert) {
         (Some(prikey), Some(pubkey_cert)) => {
-            let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls())?;
+            let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls())
+                                .map_err(|e| RestfulError::OpenSSL(e.to_string()))?;
 
             let prikey = tokio::fs::read(prikey)
                 .await
-                .context("read HTTPS private key")?;
+                .map_err(|e| RestfulError::ReadKey(e.to_string()))?;
```

Comment:
I don't think we need to explicity map the errors, shouldn't a entry in the enum be sufficient?

```rust
#[error("io error")]
IO(#[from] std::io::Error) 
```

similarly for the other errors, we can look at the error types that the fn return and wrap them like this.